File: src/my_project/medium_problems/from51to100/max_area_log_n.py
from typing import List 

class Solution:
    def maxArea(self, height: List[int]) -> int:
        
        l = 0
        r = len(height) -1
        area = 0

        while l < r:
            # Calculating the max area
            area = max(area, min(height[l],
                            height[r]) * (r - l))

            if height[l] < height[r]:
                l += 1
            else:
                r -= 1
        return area

File: src/my_project_typescript/medium_problems_typescript/from1to50/maxAreaLogn.ts
function maxArea(height: number[]): number {

    let l: number = 0
    let r: number = height.length -1 
    let area: number = 0

    while (l < r){
        area = Math.max(area, Math.min(height[l], height[r])*(r-l))

        if (height[l] < height[r]){
            l += 1
        }else {
            r-=1
        }
    }

    return area

};